geode-core/src/main/java/org/apache/geode/management/internal/cli/result/model/TabularResultModel.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.management.internal.cli.result.model;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.geode.management.internal.cli.result.ResultData;
+
+public class TabularResultModel extends AbstractResultModel {
+
+  /*
+   * Table data mapped by column name. The map needs to be able to maintain insertion order.
+   */
+  private Map<String, List<String>> table = new LinkedHashMap<>();
+
+  public TabularResultModel() {}
+
+  public String getType() {
+    return ResultData.TYPE_TABULAR;
+  }
+
+  public void setType(String type) {
+    if (!ResultData.TYPE_TABULAR.equals(type)) {
+      throw new IllegalArgumentException("Trying to set a table to type: " + type);
+    }
+  }
+
+  public TabularResultModel accumulate(String column, String value) {
```

Comment:
in addition to this accumulate method, I would love to see methods like:
tabularResultModel.setHeader(header1, header2, header3....)
tabularResultModel.addRow(value1, value2, value3....)